pkg/kube/kclient/delayed.go

```diff
@@ -86,14 +102,20 @@ func (s *delayedClient[T]) ListUnfiltered(namespace string, selector klabels.Sel
 	return nil
 }
 
-func (s *delayedClient[T]) AddEventHandler(h cache.ResourceEventHandler) {
+func (s *delayedClient[T]) AddEventHandler(h cache.ResourceEventHandler) cache.ResourceEventHandlerRegistration {
 	if c := s.inf.Load(); c != nil {
-		(*c).AddEventHandler(h)
-	} else {
-		s.hm.Lock()
-		defer s.hm.Unlock()
-		s.handlers = append(s.handlers, h)
+		return (*c).AddEventHandler(h)
 	}
+	s.hm.Lock()
+	defer s.hm.Unlock()
+
+	hasSynced := delayedHandlerRegistration{hasSynced: new(atomic.Pointer[func() bool])}
+	hasSynced.hasSynced.Store(ptr.Of(s.delayed.HasSynced))
+	s.handlers = append(s.handlers, delayedHandler{
+		ResourceEventHandler: h,
+		hasSynced:            hasSynced,
```

Comment:
what is the `hasSynced` of delayedHandler used for? I cannot see the caller

Reply:
L54